File: sdk/.net/ardvro.sdk.net/component/sqljson/controller/DbSyncController.cs
namespace ardvro.sdk.net.component.sqljson.controller;

using ardvro.core.ext.connection;

public class DbSyncController
{
    private IConnection _conection;

    public DbSyncController(IConnection connection)
    {
        _conection = connection;
    }

    public void SyncId(Object id, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/dbsync/SyncId";
        _conection.Submit(new Object[]{ id }, functionName, onResponse);
    }

    public void SyncName(String name, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/dbsync/SyncName";
        _conection.Submit(new Object[]{ name }, functionName, onResponse);
    }

    public void SyncCustom(String dbSyncType, String sourceTable, String sourceColumns, String flagSourceColumns, String wherePredicatesSource,
                           String sourceCredentials, String targetTable, String targetColumns, String targetCredentials, String sourceSqlCustom,
                           String targetSqlCustom, Action<int> onResponse)
    {
        String functionName = "ardvro/component/sqljson/dbsync/SyncCustom";
        _conection.Submit(new Object[]{ dbSyncType, sourceTable, sourceColumns, flagSourceColumns, wherePredicatesSource, sourceCredentials, targetTable, targetColumns, targetCredentials, sourceSqlCustom, targetSqlCustom }, functionName, onResponse);
    }


}

File: sdk/web/component/sqljson/controller/DbSyncController.js
var DbSyncController = function DbSyncController(cfg)
{
    let ctrl = {};

    ctrl.SyncId = function (id, callback)
    {
        cfg.Connector.Submit(id, "ardvro/component/sqljson/dbsync/syncid", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SyncName = function (name, callback)
    {
        cfg.Connector.Submit(name, "ardvro/component/sqljson/dbsync/syncname", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.SyncCustom = function (dbSyncType, sourceTable, sourceColumns, flagSourceColumns, wherePredicatesSource, sourceCredentials, targetTable, targetColumns, targetCredentials, sourceSqlCustom, targetSqlCustom, callback)
    {
        let args = [dbSyncType, sourceTable, sourceColumns, flagSourceColumns, wherePredicatesSource, sourceCredentials, targetTable, targetColumns, targetCredentials, sourceSqlCustom, targetSqlCustom];
        cfg.Connector.Submit(args, "ardvro/component/sqljson/dbsync/synccustom", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
};